### PR TITLE
⭐️  Added ReportSubmissionPolicy resources to asses Defender (MS365)

### PR DESCRIPTION
**Pull request by HRouhani**
Screenshot from 2024-05-17 15-54-20

### CODE REVIEW

File: providers/ms365/resources/ms365.lr [hunk 1]
```diff
@@ -402,6 +404,27 @@ private ms365.exchangeonline.teamsProtectionPolicy {
...
+  reportNotJunkAddresses []string
+  // Addresses to report phish
+  reportPhishAddresses []string
+  // Whether chat message reporting is enabled
+  reportChatMessageEnabled bool
+  // Whether to report chat messages to a customized address
+  reportChatMessageToCustomizedAddressEnabled bool
+}
+
+
```
Comment (tas50) [inline providers/ms365/resources/ms365.lr]:
```suggestion
```

File: providers/ms365/resources/ms365_exchange.go [hunk 2]
```diff
@@ -184,6 +198,32 @@ func convertTeamsProtectionPolicy(r *mqlMs365Exchangeonline, data []*TeamsProtec
 	return plugin.TValue[[]interface{}]{Data: result, State: plugin.StateIsSet, Error: teamsProtectionPolicyErr}
 }
 
+// Related to ReportSubmissionPolicy as a seperate function
+func convertReportSubmissionPolicy(r *mqlMs365Exchangeonline, data []*ReportSubmissionPolicy) plugin.TValue[[]interface{}] {
```
Comment (preslavgerchev) [inline providers/ms365/resources/ms365_exchange.go]: id prefer if we have those func convert to the MQL resource instead. we can leave the setting of  `plugin.TValue` to the caller of this function

Comment (HRouhani) [reply providers/ms365/resources/ms365_exchange.go]: Ok sure, done, Thanks

File: providers/ms365/resources/ms365.lr.manifest.yaml [hunk 3]
```diff
@@ -260,6 +260,8 @@ resources:
       owaMailboxPolicy: {}
       phishFilterPolicy: {}
       remoteDomain: {}
+      reportSubmissionPolicies:
+        min_mondoo_version: latest
```
Comment (tas50) [inline providers/ms365/resources/ms365.lr.manifest.yaml]:
```suggestion
        min_mondoo_version: 9.0.0
```

Comment (HRouhani) [reply providers/ms365/resources/ms365.lr.manifest.yaml]: Done, Thanks

**Review by preslavgerchev — APPROVED** [review]

**Pull request merged**